modules/develop/pages/data-transforms/build.adoc

```diff
@@ -0,0 +1,655 @@
+
+Redpanda tracks the offsets of the records processed by transform functions. If an error escapes the Wasm virtual machine (VM), the VM will fail. When the Wasm engine detects this failure and starts a new VM, the transform function will try to process the input topics from the last processed offset again, leading to the same error and causing the function to be stuck in an error loop.
+
+To prevent this behavior, log errors and take appropriate action without stopping the processing of records. This approach ensures that your transform function can continue to operate even when encountering problematic records.
+
+In this example, an error will be logged if a record does not contain a key and that record will not be processed. Subsequent records will continue to be processed.
+
+[tabs]
+======
+Go::
++
+--
+[source,go]
+----
+package main
+
+import (
+    "log"
+    "github.com/redpanda-data/redpanda/src/transform-sdk/go/transform"
+)
+
+func main() {
+    transform.OnRecordWritten(handleRecord)
+}
+
+func handleRecord(event transform.WriteEvent, writer transform.RecordWriter) error {
+  record := event.Record()
+  if record.Key == nil {
+    // Log the error and continue processing
+    log.Println("Error: Record key is nil")
+    // Do not return the error
+    return nil
+  }
+
+  err := writer.Write(record)
+  if err != nil {
+    // Log the error and continue processing
+    log.Println("Error writing record:", err)
+  }
+  // Ensure no error is returned
+  return nil
+}
+----
+--
+Rust::
++
+--
+[source,rust]
+----
+use redpanda_transform_sdk::*;
+use log::error;
+
+fn main() {
+  // Set up logging
+  env_logger::init();
+  on_record_written(handle_record);
+}
+
+fn handle_record(event: WriteEvent, writer: &mut RecordWriter) -> anyhow::Result<()> {
+  let record = event.record;
+  if record.key().is_none() {
+    // Log the error and continue processing
+    error!("Error: Record key is nil");
+    // Do not return the error
+    return Ok(());
+  }
+
+  if let Err(err) =  writer.write(record) {
+    // Log the error and continue processing
+    error!("Error writing record: {}", err);
+  }
+  // Ensure no error is returned
+  Ok(())
+}
+----
+--
+======
+
+When you deploy this transform function, and produce a message without a key, you'll get the following in the logs:
+
+[source,js,role="no-copy"]
+----
+{
+  "body": {
+    "stringValue": "2024/06/20 08:17:33 Error: Record key is nil\n"
+  },
+  "timeUnixNano": 1718871455235337000,
+  "severityNumber": 13,
+  "attributes": [
+    {
+      "key": "transform_name",
+      "value": {
+        "stringValue": "test"
+      }
+    },
+    {
+      "key": "node",
+      "value": {
+        "intValue": 0
+      }
+    }
+  ]
+}
+----
```

Comment:
This is a good opportunity to showcase `rpk transform logs`, because we don't tell people how to see this. Using the rpk command we can show people the command that generates this.